File: 2728-sum-in-a-matrix/solution.ts
function matrixSum(nums: number[][]): number {
  nums.forEach((n) => n.sort((a, b) => b - a));
  const m = nums[0].length;
  const newNums = [...Array(m).keys()].map((i) => nums.map((n) => n[i]));
  newNums.forEach((n) => n.sort((a, b) => b - a));
  return newNums.reduce((acc, cur) => acc + cur[0], 0);
}

File: 2728-sum-in-a-matrix/solution.cpp
class Solution {
public:
    int matrixSum(vector<vector<int>>& nums) {
        int ans = 0, m = nums.size(), n = nums[0].size();
        for(auto& num : nums) sort(num.rbegin(), num.rend());
        for(int j = 0; j < n; j++) {
            int mx = -1;
            for(int i = 0; i < m; i++) mx = max(mx, nums[i][j]);
            ans += mx;
        }
        return ans;
    }
};
